cli/commands/dump/dump.go

```diff
@@ -0,0 +1,155 @@
+	"github.com/sensu/sensu-go/cli/commands/helpers"
+	"github.com/spf13/cobra"
+)
+
+const (
+	// VerbList is the sensuctl verb to list resources.
+	VerbList = "list"
+	// VerbInfo is the sensuctl verb to get info about a resource.
+	VerbInfo = "info"
+)
+
+// Action is a resource/verb tuple for sensuctl commands.
+type Action struct {
+	Resource   string
+	Verb       string
+	Namespaced bool
+}
+
+var (
+	// All is all the core resource types and associated sensuctl verbs (non-namespaced resources are intentionally ordered first).
+	All = []Action{
+		Action{Resource: "namespace", Verb: VerbList, Namespaced: false},
+		Action{Resource: "cluster-role", Verb: VerbList, Namespaced: false},
+		Action{Resource: "cluster-role-binding", Verb: VerbList, Namespaced: false},
+		Action{Resource: "user", Verb: VerbList, Namespaced: false},
+		Action{Resource: "tessen", Verb: VerbInfo, Namespaced: false},
+		Action{Resource: "asset", Verb: VerbList, Namespaced: true},
+		Action{Resource: "check", Verb: VerbList, Namespaced: true},
+		Action{Resource: "entity", Verb: VerbList, Namespaced: true},
+		Action{Resource: "event", Verb: VerbList, Namespaced: true},
+		Action{Resource: "filter", Verb: VerbList, Namespaced: true},
+		Action{Resource: "handler", Verb: VerbList, Namespaced: true},
+		Action{Resource: "hook", Verb: VerbList, Namespaced: true},
+		Action{Resource: "mutator", Verb: VerbList, Namespaced: true},
+		Action{Resource: "role", Verb: VerbList, Namespaced: true},
+		Action{Resource: "role-binding", Verb: VerbList, Namespaced: true},
+		Action{Resource: "silenced", Verb: VerbList, Namespaced: true},
+	}
+)
+
+// Command dumps generic Sensu resources to a file or STDOUT.
+func Command(cli *cli.SensuCli) *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "dump [RESOURCE TYPE],[RESOURCE TYPE]... [-f FILE]",
+		Short: "dump resources to a file or STDOUT",
+		RunE:  execute(cli),
+	}
+
+	helpers.AddFormatFlag(cmd.Flags())
+	helpers.AddAllNamespace(cmd.Flags())
+	_ = cmd.Flags().StringP("file", "f", "", "file to dump resources to")
+
+	return cmd
+}
+
+func execute(cli *cli.SensuCli) func(*cobra.Command, []string) error {
+	return func(cmd *cobra.Command, args []string) error {
+		if len(args) != 1 {
+			_ = cmd.Help()
+			return errors.New("invalid argument(s) received")
+		}
+
+		// get the configured format or the flag override
+		format := cli.Config.Format()
+		if flag := helpers.GetChangedStringValueFlag("format", cmd.Flags()); flag != "" {
+			format = flag
+		}
+		switch format {
+		case "yaml", "wrapped-json":
+		default:
+			format = "yaml"
+		}
+
+		// parse the comma separated resource types and match against the defined actions
+		var actions []Action
+		if args[0] == "all" {
+			actions = All
+		} else {
+			types := strings.Split(args[0], ",")
+			for _, t := range types {
+				length := len(actions)
+				for _, action := range All {
+					if t == action.Resource {
+						actions = append(actions, action)
```

Comment:
We should probably make sure we don't add duplicates into this slice, otherwise right now if I do `sensuctl dump check,check`, it duplicates the resources printed. I could see this being problematic when you want to dump all resources except one and the list of types become long and therefore easy to make a mistake and duplicate a type.

Reply:
Good catch!

Reply:
In the event of duplicate resource types, the command returns the following error: `Error: duplicate resource type: check`